### ISSUE TITLE
assert that "TaskName" in JSON files corresponds to actual tasks

### ISSUE THREAD

**Issue by sappelhoff** [bids-standard/bids-validator#1033]
e.g., This file: https://github.com/bids-standard/bids-examples/blob/7dc73456fe41bf204c1c71f6512293d2f33c6686/eeg_matchingpennies/task-matchingpennies_eeg.json#L2

has as task name `"Matching Pennies"` although the label in the files is `task-matchingpennies`

I think the validator should ensure that these two "fields" match

**Comment by rwblair** [comment]
From the spec:
"The task label included in the file name is derived from this TaskName field by removing all non-alphanumeric ([a-zA-Z0-9]) characters. For example TaskName faces n-back will correspond to task label facesnback"

From that I take TaskName as being the generally accepted, more human readable, name of the task with the entity for the task name being properly defined in the filename itself.

The appropriate check might be to take TaskName, strip non-alphanum (convert to lowercase?) and compare that with the found task entity.

**Comment by sappelhoff** [comment]
> The appropriate check might be to take TaskName, strip non-alphanum (convert to lowercase?) and compare that with the found task entity.

agreed! except for the "convert to lowercase", because I think casing is not touched by the spec.

**Comment by effigies** [comment]
Not reading this as REQUIRED language, so if we do this, I think it should be a warning. It doesn't feel extremely helpful, though.
